wp-includes/sqlite/class-wp-sqlite-translator.php

```diff
@@ -272,6 +277,48 @@ class WP_SQLite_Translator {
 	 */
 	private $last_reserved_keyword;
 
+	/*
+	 * True if a VACUUM operation should be done on shutdown, to handle OPTIMIZE TABLE and similar operations.
+	 *
+	 * @var bool
+	 */
+	private $vacuum_requested = false;
+
+	/**
+	 * True if the present query is metadata
+	 *
+	 * @var bool
+	 */
+	private $is_information_schema_query = false;
+
+	/**
+	 * True if a GROUP BY clause is detected.
+	 *
+	 * @var bool
+	 */
+	private $has_group_by = false;
+
+	/**
+	 * 0 if no LIKE is in progress, otherwise counts nested parentheses.
+	 *
+	 * @var int
+	 */
+	private $like_expression_nesting = 0;
```

Comment:
A generic stack of expression would scale better. There's already a `call_stack` in WP_SQLite_Query_Rewriter and I worry we may eventually end up with many different ways of tracking nesting level (like  `like_expression_nesting`, `case_expression_nesting` etc). Not a blocker here, but let's keep an eye on how this code evolves.

Reply:
Added a @todo.